tests/mlflow/tensorflow/test_tensorflow_model_export.py

```diff
@@ -55,32 +81,126 @@ def test_log_saved_model(self):
-                xpred = x.predict(pandas.DataFrame(data=self._X, columns=self._feature_names))
-                saved = []
-                for s in self._dnn_predict:
-                    saved.append(s['predictions'])
-                loaded = []
-                for index, rows in xpred.iterrows():
-                    loaded.append(rows)
+
+                results = self.helper(feature_spec, tmp, dnn, dnn_predict, pandas.DataFrame(data=X, columns=feature_names))
+
+                # Asserting that the loaded model predictions are as expected.
+                np.testing.assert_array_equal(results[0], results[1])
+            finally:
+                # Restoring the old logging location.
+                tracking.end_run()
+                tracking.set_tracking_uri(old_tracking_dir)
+
+
+    def test_cat_cols(self):
+        """
+        This tests logging capabilities on datasets with categorical columns.
+        See https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/get_started/regression/imports85.py
+        for reference code.
+        """
+        with TempDir(chdr=False, remove_on_exit=True) as tmp:
+            # Downloading the data into a pandas DataFrame.
+            URL = "https://archive.ics.uci.edu/ml/machine-learning-databases/autos/imports-85.data"
+            path = tf.contrib.keras.utils.get_file(URL.split("/")[-1], URL)
+            # Order is important for the csv-readers, so we use an OrderedDict here.
+            defaults = collections.OrderedDict([
+                ("symboling", [0]),
+                ("normalized-losses", [0.0]),
+                ("make", [""]),
+                ("fuel-type", [""]),
+                ("aspiration", [""]),
+                ("num-of-doors", [""]),
+                ("body-style", [""]),
+                ("drive-wheels", [""]),
+                ("engine-location", [""]),
+                ("wheel-base", [0.0]),
+                ("length", [0.0]),
+                ("width", [0.0]),
+                ("height", [0.0]),
+                ("curb-weight", [0.0]),
+                ("engine-type", [""]),
+                ("num-of-cylinders", [""]),
+                ("engine-size", [0.0]),
+                ("fuel-system", [""]),
+                ("bore", [0.0]),
+                ("stroke", [0.0]),
+                ("compression-ratio", [0.0]),
+                ("horsepower", [0.0]),
+                ("peak-rpm", [0.0]),
+                ("city-mpg", [0.0]),
+                ("highway-mpg", [0.0]),
+                ("price", [0.0])
+            ])
+
+            types = collections.OrderedDict((key, type(value[0]))
+                                            for key, value in defaults.items())
+            df = pandas.read_csv(path, names=types.keys(), dtype=types, na_values="?")
+            df = df.dropna()
+            # Keeping only the data we are testing.
+            df = df[["price", "body-style", "curb-weight", "highway-mpg"]]
```

Comment:
If we're only keeping these columns, do we need to have all the columns in `defaults` above?